source/common/router/router_ratelimit.cc

```diff
@@ -241,6 +241,37 @@ bool HeaderValueMatchAction::populateDescriptor(RateLimit::DescriptorEntry& desc
   }
 }
 
+QueryParameterValueMatchAction::QueryParameterValueMatchAction(
+    const envoy::config::route::v3::RateLimit::Action::QueryParameterValueMatch& action)
+    : descriptor_value_(action.descriptor_value()),
+      descriptor_key_(!action.descriptor_key().empty() ? action.descriptor_key() : "query_match"),
+      expect_match_(PROTOBUF_GET_WRAPPED_OR_DEFAULT(action, expect_match, true)),
+      action_query_parameters_(buildQueryParameterMatcherVector(action)) {}
+
+bool QueryParameterValueMatchAction::populateDescriptor(
+    RateLimit::DescriptorEntry& descriptor_entry, const std::string&,
+    const Http::RequestHeaderMap& headers, const StreamInfo::StreamInfo&) const {
+  Http::Utility::QueryParams query_parameters =
+      Http::Utility::parseQueryString(headers.getPathValue());
```

Comment:
nit: should we use `parseAndDecodeQueryString` here to decode escaped parameter value?

Reply:
yes, i think `parseAndDecodeQueryString` is more appropriate, thanks.